mlcg/nn/prior.py

```diff
@@ -1065,3 +1065,233 @@ def from_user(*args):
+        ).t()
+
+        x0s = torch.vstack(
+            [self.x0s[ii][interaction_types] for ii in range(self.n_degs - 1)]
+        ).t()
+        v_0s = self.v_0[interaction_types].t()
+        return {"ks": ks, "x0s": x0s, "v_0s": v_0s}
+
+    def forward(self, data):
+        mapping_batch = data.neighbor_list[self.name]["mapping_batch"]
+        features = self.data2features(data).flatten()
+        params = self.data2parameters(data)
+        V0s = params["v_0s"].t()
+        ks = params["ks"].t()
+        x0s = params["x0s"].t()
+        y = Quartic.compute(
+            features,
+            ks,
+            V0s,
+            x0s,
+        )
+        y = scatter(y, mapping_batch, dim=0, reduce="sum")
+        data.out[self.name] = {"energy": y}
+        return data
+
+    @staticmethod
+    def compute(
+        x: torch.Tensor, ks: torch.Tensor, V0: torch.Tensor, x0s: torch.Tensor
+    ):
+        """Quartic potential interaction with missing linear term.
+
+        .. math:
+
+            V(r) = V0 + \sum_{n=2}^{4} k_n (x-x_n)^n
+
+        """
+        V = 0
+        for i in range(3):
+            V += ks[i] * (x - x0s[i]) ** (i + 2)
+
+        V += V0
+        return V
+
+    @staticmethod
+    def _quartic_model(x, a, b, c, d, xa, xb, xc):
+        return a * (x - xa) ** 2 + b * (x - xb) ** 3 + c * (x - xc) ** 4 + d
+
+    @staticmethod
+    def _init_quartic_parameters(n_degs):
+        """
+        Helper method for guessing initial parameter values
+        Not used for now
+        """
+        ks = [1.0 for _ in range(n_degs - 1)]
+        x0s = [0.0 for _ in range(n_degs - 1)]
+        V0 = -1.0
+        p0 = [V0]
+        p0.extend(ks)
+        p0.extend(x0s)
+        return p0
+
+    @staticmethod
+    def _init_quartic_parameter_dict(n_degs):
+        """Helper method for initializing the parameter dictionary"""
+        stat = {"ks": {}, "x0s": {}, "v_0": 0.0}
+        k_names = ["k_" + str(ii) for ii in range(2, n_degs + 1)]
+        x0_names = ["x0_" + str(ii) for ii in range(2, n_degs + 1)]
+        for ii in range(n_degs - 1):
+            k_name = k_names[ii]
+            x0_name = x0_names[ii]
+            stat["ks"][k_name] = {}
+            stat["x0s"][x0_name] = {}
+        return stat
+
+    @staticmethod
+    def _make_quartic_dict(stat, popt, n_degs):
+        """Helper method for constructing a fitted parameter dictionary"""
+        stat["v_0"] = popt[0]
+        k_names = sorted(list(stat["ks"].keys()))
+        x0_names = sorted(list(stat["x0s"].keys()))
+        for ii in range(n_degs - 1):
+            k_name = k_names[ii]
+            x0_name = x0_names[ii]
+            stat["ks"][k_name] = popt[ii]
+            stat["x0s"][x0_name] = popt[ii + n_degs]
+
+        return stat
+
+    @staticmethod
+    def fit_quartic_from_potential_estimates(
+        bin_centers_nz: torch.Tensor,
+        dG_nz: torch.Tensor,
+        **kwargs,
+    ):
+        """
```

Comment:
You need to make a PR also to mlcg-tk because we need to call also the fitting function from there.